pkg/cidata/cidata.TEMPLATE.d/user-data

```diff
@@ -84,6 +84,12 @@ ca_certs:
 bootcmd:
   {{- range $cmd := $.BootCmds }}
 - |
+  # We need to embed the params.env as a here-doc because /mnt/lima-cidata is not yet mounted
+  while read -r line; do [ -n "$line" ] && export "$line"; done <<'EOF'
+    {{- range $key, $val := $.Param }}
+  PARAM_{{ $key }}={{ $val }}
```

Comment:
There is the pesky issue of how to do the quoting if `$val` contains quotes. This will work with the current approach:

```
PARAM_FOO=foo"bar
PARAM_BAR=foo'bar
```

How would you generate the correct code via the template?

The other argument would be that the code is mostly the same as the code used elsewhere to process `lima.env` and `param.env`.

Reply:
I feel that value should be escaped with `shellescape.Quote()` both when creating `param.env` and when expanding into `user-data`. Essentially, what the `while read -r...` loop is doing is escaping, right? In `.env` files as well, it's common to quote the value on the right-hand side of the `=`. 🤔

Reply:
> I feel that value should be escaped with `shellescape.Quote()` both when creating `param.env` and when expanding into `user-data`.

I consider `lima.env` (and now `param.env`) to be similar to `/etc/environment`. They are not shell scripts and don't follow shell syntax; they just list key/value pairs.

Maybe we can change them to be shell scripts instead, but there is the risk that we may be overlooking some edge case for which it won't work. Why change code that seems to have been working correctly for us for a long time? What is the benefit to offset the risk?

Reply:
> I consider `lima.env` (and now `param.env`) to be similar to `/etc/environment`

Got it, I understand with this perspective. Thank you!